### PR TITLE
Editor: Don't show preview on publish when private is selected

### PR DESCRIPTION
**Pull request by jonathansadowski**
This PR modifies the behavior of the post-publish preview such that it is no longer shown when a post is published as a result of "private" being selected as post-visibility from the post-settings sidebar.

If, however, "private" is selected from the confirmation sidebar, the preview will continue to be shown.

This PR is part of a larger epic. See p8F9tW-3F-p2.

***To test:***
* Checkout branch
* Run the branch with previews enabled: `ENABLE_FEATURES=post-editor/delta-post-publish-preview make run`
* Draft a post, and from post-settings change the post visibility to "private".
* From the ensuing confirmation dialog, select "yes". Your post should be saved as private and you **should not** be shown a preview.
* Draft another post. This time save it as "public" or "password protected". Upon publishing the post, you **should** be shown a preview.
* Stop the branch.
* Run the branch with previews **and** the confirmation flow enabled: `ENABLE_FEATURES=post-editor/delta-post-publish-preview,post-editor/delta-post-publish-flow make run`
* Change visibility of a new post to private from post-settings as above. Again your post should be published as private and again you should _not_ see a preview.
* Create a new post. This time change it's visibility to private from the confirmation sidebar.
* After the post is published as private from the confirmation sidebar, you should see a preview of your post.
* Publishing posts as "public" or "password protected" should result in a preview regardless of whether the change was made from the confirmation sidebar or post-settings.

### CODE REVIEW

**Comment by mjuhasz** [comment]
Based on my testing, the preview keeps showing even when the post is published privately.

File: client/post-editor/post-editor.jsx [hunk 1]
```diff
@@ -820,6 +816,11 @@ export const PostEditor = React.createClass( {
 
 	onSaveSuccess: function( message, action, link ) {
 		const post = PostEditStore.get();
+		const isNotPrivateOrIsConfirmed = ( 'private' !== post.status ) || ( 'closed' !== this.state.confirmationSidebar );
```
Comment (michaeldcain) [inline client/post-editor/post-editor.jsx]: It looks like `onPublish()` is setting the `confirmationSidebar` state to `publish` before this check fires - thus always true.

Comment (jonathansadowski) [inline client/post-editor/post-editor.jsx]: Thanks. It looks like we probably shouldn't be entering "publishing" state for the confirmation sidebar if we weren't in "open" state to begin with -- so I fixed this by preventing that state from being reached.

**Comment by jonathansadowski** [comment]
Good catch @mjuhasz and @michaeldcain! The combination of feature flags caused me to miss that. I've updated the PR and that should be fixed now.

**Comment by mjuhasz** [comment]
It works as expected now 👍

**Pull request merged**